core/src/main/scala/cats/instances/queue.scala

```diff
@@ -17,6 +19,20 @@ trait QueueInstances extends cats.kernel.instances.QueueInstances {
 
       def combineK[A](x: Queue[A], y: Queue[A]): Queue[A] = x ++ y
 
+      override def combineAllOptionK[A](as: IterableOnce[Queue[A]]): Option[Queue[A]] = {
+        val iter = as.iterator
+        if (iter.isEmpty) None else Some(appendAll(iter, Queue.newBuilder[A]).result())
+      }
+
+      override def fromIterableOnce[A](as: IterableOnce[A]): Queue[A] = {
+        val iter = as.iterator
+        val builder = Queue.newBuilder[A]
```

Comment:
Won't `builder ++= as` work here?

Reply:
I wonder if this is worth the micro optimization of `if (as.isEmpty) Queue.empty else { ... }` since in that case we don't allocate the builder?